iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/execution/executor/RegionWriteExecutor.java

```diff
@@ -241,22 +238,25 @@ public RegionExecutionResult visitPipeEnrichedInsertNode(
 
     private RegionExecutionResult executeDataInsert(
         InsertNode insertNode, WritePlanNodeExecutionContext context) {
-      RegionExecutionResult response = new RegionExecutionResult();
+      if (context.getRegionWriteValidationRWLock() == null) {
+        String message = "Failed to get the lock of the region because the region is not existed.";
+        return RegionExecutionResult.create(
+            false, message, RpcUtils.getStatus(TSStatusCode.WRITE_PROCESS_ERROR, message));
+      }
+
```

Comment:
Is it necessary to set `needRetry` field?

Reply:
Not necessary, `needRetry` field is for read retry and is not required for write.

Reply:
@HxpSerein Then maybe rename it to `readNeedRetry` ?

Reply:
Good point. Fixed.